contracts/contracts/StealthKeyRegistry.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.7;
+
+contract StealthKeyRegistry {
+  // =========================================== Events ============================================
+
+  /// @dev Event emitted when a user updates their resolver stealth keys
+  event StealthKeyChanged(
+    address indexed registrant,
+    uint256 spendingPubKeyPrefix,
+    uint256 spendingPubKey,
+    uint256 viewingPubKeyPrefix,
+    uint256 viewingPubKey
+  );
+
+  // ======================================= State variables =======================================
+
+  /// @dev The domain typehash used for EIP-712 signatures in setStealthKeysOnBehalf
+  bytes32 public constant EIP712DOMAIN_TYPEHASH =
+    keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
```

Comment:
Any reason this is pulled out as a constant? I think it's typically inlined in the `DOMAIN_SEPARATOR` calculation in the constructor. Mostly a style thing but might be good to be consistent. See [Dai](https://github.com/makerdao/dss/blob/81441305eebb80e2bca39afe33c9ef44446f1fb2/src/dai.sol#L64), [Uniswap V2](https://github.com/Uniswap/uniswap-v2-core/blob/4dd59067c76dea4a0e8e4bfdda41877a6b16dedc/contracts/UniswapV2ERC20.sol#L31)